polymath/extensions/shaper.py

```
@@ -3,383 +3,391 @@
-    else:
-        dtype = np.bool_
-
-    values = np.empty((len(args),) + np.shape(args[subclass_indx]._values), dtype=dtype)
-    for i in range(len(args)):
-        if args[i] is None:
-            values[i] = 0
-        else:
-            values[i] = args[i]._values
+    @staticmethod
+    @staticmethod
+    def stack(*args, recursive=True):
```

Ruff (0.14.7)

Missing return type annotation for staticmethod `stack`

(ANN205)